Add guide to blog contributions (#50092)
kubernetes/website: content/en/docs/contribute/blog/guidelines.md

---
title: Blog guidelines
content_type: concept
weight: 40
---

<!-- overview -->

> These guidelines cover the main [Kubernetes blog](/blog/) and the Kubernetes

Author: I had it like that and then changed my mind. Will change it back.

[contributor blog](https://k8s.dev/blog/).

Read [contributing to Kubernetes blogs](/docs/contribute/blog/) to learn about
the two official blogs and the differences between them.

All blog content must also adhere to the overall policy in the
[content guide](/docs/contribute/style/content-guide/).

## Original content

> The Kubernetes project accepts **original content only**, in English.

Author: No; the point I want to make is that PRs to add a new article must be in English to be accepted.

---
title: Blog guidelines
content_type: concept
weight: 40
---

<!-- overview -->

These guidelines cover the main Kubernetes blog and the Kubernetes
contributor blog.

All blog content must also adhere to the overall policy in the
[content guide](/docs/contribute/style/content-guide/).

> # {{% heading "prerequisites" %}}

Reviewer: Is there a mistake here, or is the single `#` character used on purpose?

or published outside of the Kubernetes project.
{{< /note >}}

The official Kubernetes blogs are not the place for vendor pitches or for articles that promote
a specific solution from outside Kubernetes.

This restriction even carries across to promoting other Linux Foundation and CNCF projects.
Many CNCF projects have their own blog. These are often a better choice for posts about a specific
project, even if that other project is designed specifically to work with Kubernetes (or with Linux,
etc).

Articles must contain content that applies broadly to the Kubernetes community. For example, a
submission should focus on upstream Kubernetes as opposed to vendor-specific configurations.
Hyperlinks in articles should primarily be to the official Kubernetes documentation. When using external
references, links should be diverse - For example a submission shouldn't contain only links
back to a single company's blog.

Sometimes this is a delicate balance. You can ask in Slack ([#sig-docs-blog](https://kubernetes.slack.com/archives/CJDHVD54J))
for guidance on whether a post is appropriate for the Kubernetes blog and / or contributor blog -
don't hesitate to reach out

> The [content guide](/docs/contribute/style/content-guide/) applies unconditionally to blog articles

Author: I made some changes. However, I actually think we'd be better off renaming https://kubernetes.io/docs/contribute/style/content-guide/ and calling it a _content policy_.

and the PRs that add them. Bear in mind that some restrictions in the guide state that they are only relevant to documentation; those marked restrictions don't apply to blog articles.

> You must write [original content](#original-content) and you must have permission to license that content to the

Author: I reworded to provide an example.

> You must write [original content](#original-content) and you must have permission to license that content to the

Author: (how's this)

Cloud Native Computing Foundation (so that the Kubernetes project can legally publish it).

The website is localized into many languages; English is the “upstream” for all the other
localizations. Even if you yourself speak another language and would be happy to provide a localization,
that should be in a separate pull request (see [languages per PR](/docs/contribute/new-content/#languages-per-pr)).

## Blog content guidance {#what-we-publish}

### Content examples

Here are some examples of content that is appropriate for the main Kubernetes blog:

* Announcements about new Kubernetes capabilities
* Explanations of how to achieve an outcome using Kubernetes; for example, tell us about your
  low-toil improvement on the basic idea of a rolling deploy
* Comparisons of different software options that have a link to Kubernetes and cloud native. It's
  OK to have a link to one of these options so long as you fully disclose your conflict of
  interest / relationship.
* Stories about problems or incidents, and how you resolved them

> * Articles discussing building a cloud-native platform for specific use cases

Reviewer 2: ```suggestion
* Articles discussing building a cloud native platform for specific use cases
```

  your opinion about the good or bad points about Kubernetes
* Announcements and stories about non-core Kubernetes, such as the Gateway API
* [Post-release announcements and updates](#post-release-comms)
* Messages about important Kubernetes security vulnerabilities
* Kubernetes projects updates
* Tutorials and walkthroughs
* Thought leadership around Kubernetes and cloud native
* The components of Kubernetes are purposely modular, so writing about existing integration
  points like CNI and CSI are on topic. Provided you don't write a vendor pitch, you can also write
  about what is on the other end of these integrations.

Here are some examples of content that is appropriate for the Kubernetes contributor blog:

* Articles about how to test your change to Kubernetes code
* Content around non-code contribution
* Discussions about alpha features where the design is still under discussion

> * "Meet the team" articles about working groups, special interest groups etc

Reviewer 2: ```suggestion
* "Meet the team" articles about working groups, special interest groups, etc.
```

### Other considerations {#content-considerations-other}

The Kubernetes website has an ICP licence from the government of China. It's unlikely to be a problem but we also cannot publish articles that would be blocked by the Chinese government's official filtering of internet content.

Topics related to participation in or results of Kubernetes SIGs activities are always on
topic (see the work in the [Contributor Comms Team](https://github.com/kubernetes/community/blob/master/communication/contributor-comms/blogging-resources/blog-guidelines.md#contributor-comms-blog-guidelines)
for support on these posts).

The project typically mirrors these articles to both blogs.

- Blog posts should be original content

  - The official blog is not for repurposing existing content from a third party as new content.
  - The [license](https://github.com/kubernetes/website/blob/main/LICENSE) for the blog allows
    commercial use of the content for commercial purposes, but not the other way around.

- Blog posts should aim to be future proof

  - Given the development velocity of the project, we want evergreen content that won't require

>     updates to stay accurate for the reader. 

Reviewer 2: Because this is for a blog, I thought we wouldn't need to always consider how evergreen it should be? (And there is a separate frontmatter support for the evergreen articles)

Author: ```suggestion
  - Given the development velocity of the project, we prefer timeless content that won't require
    updates to stay accurate for the reader.
```

  - It can be a better choice to add a tutorial or update official documentation than to write a
    high level overview as a blog post.
    - Consider concentrating the long technical content as a call to action of the blog post, and
      focus on the problem space or why readers should care.

## Article scheduling

> The usual process for submitting a blog article is to 

Reviewer 2: Just noting an unfinished sentence here

  [CNCF marketing](https://www.cncf.io/about/contact/) is a more appropriate choice than submitting a blog post.
- Sometimes reviews can get backed up. If you feel your review isn't getting the attention it needs,
  you can reach out to the blog team on the [`#sig-docs-blog` Slack channel](https://kubernetes.slack.com/messages/sig-docs-blog/)
  to ask in real time.

## Technical considerations for submitting a blog post

Submissions need to be in Markdown format to be used by the [Hugo](https://gohugo.io/) generator
for the blog. There are [many resources available](https://gohugo.io/documentation/) on how to use
this technology stack.

For illustrations, diagrams or charts, the [figure shortcode](https://gohugo.io/content-management/shortcodes/#figure)
can be used. For other images, we strongly encourage use of alt attributes; if an image doesn't
need any alt attrribute, maybe it's not needed in the article at all.

We recognize that this requirement makes the process more difficult for less-familiar folks to
submit, and we're constantly looking at solutions to lower this bar. If you have ideas on how to

> lower the barrier, please volunteer to help out. 

Reviewer 3: ```suggestion
lower the barrier, please volunteer to help by reaching out on the [`#sig-docs-blog` Slack channel](https://kubernetes.slack.com/messages/sig-docs-blog/).
```
